### ISSUE TITLE
put-before/put-after in commented code breaks header detection

### ISSUE THREAD

**Issue by vaclavsvejcar** [vaclavsvejcar/headroom#57]
When comment is present within the _Haddock_ header (e.g. in code block), it confuses header detection logic and _Headroom_ incorrectly strips the rest of the header below this line.

For example this:
```haskell
{-|
Module      : Data.VCS.Ignore.Repo
Description : Type class representing the VCS repository
Copyright   : (c) 2020 Vaclav Svejcar
License     : BSD-3-Clause
Maintainer  : vaclav.svejcar@gmail.com
Stability   : experimental
Portability : POSIX

This module contains /type class/ representing the selected type of /VCS/
repository.

@
  {-# LANGUAGE TypeApplications #-}
@
-}
```

Becomes this (after header replacement) - which is invalid code
```haskell
{-|
Module      : Data.VCS.Ignore.Repo
Description : Type class representing the VCS repository
Copyright   : (c) 2020 Vaclav Svejcar
License     : BSD-3-Clause
Maintainer  : vaclav.svejcar@gmail.com
Stability   : experimental
Portability : POSIX

This module contains /type class/ representing the selected type of /VCS/
repository.

@
-}

@
-}
```

**Comment by vaclavsvejcar** [comment]
Unfortunately I don't have enough time to include this to [v0.4.0.0](https://github.com/vaclavsvejcar/headroom/milestone/10) and since I don't want to block this release anymore, I postpone this for [v0.4.1.0](https://github.com/vaclavsvejcar/headroom/milestone/11) which I expect myself to release later this month.

**Comment by vaclavsvejcar** [comment]
This is more serious than expected, rescheduling this back for [v0.4.0.0](https://github.com/vaclavsvejcar/headroom/milestone/10) release.

**Comment by vaclavsvejcar** [comment]
Implemented in `master`, will be part of the [v0.4.0.0](https://github.com/vaclavsvejcar/headroom/milestone/10) release.
